tests/unit/js/window-tests.js

```diff
@@ -0,0 +1,138 @@
+/*
+Infusion-Electron Window Tests
+Copyright 2019 Colin Clark
+
+Licensed under the 3-Clause "New" BSD license.
+You may not use this file except in compliance with one these
+Licenses.
+
+You may obtain a copy of the 3-Clause "New" BSD License at
+https://github.com/colinbdclark/infusion-electron/raw/master/LICENSE.txt
+*/
+
+"use strict";
+
+var fluid = require("infusion"),
+    jqUnit = require("node-jqunit"),
+    electron = fluid.registerNamespace("electron");
+
+fluid.require("%infusion-electron/tests/unit/js/utils/test-app.js");
+fluid.require("%infusion-electron/tests/unit/js/utils/test-window.js");
+
+fluid.defaults("electron.tests.windowTestEnvironment", {
+    gradeNames: ["fluid.test.testEnvironment"],
+
+    components: {
+        app: {
+            type: "electron.tests.testApp",
+            options: {
+                components: {
+                    windough: {
+                        createOnEvent: "onReady",
+                        type: "electron.tests.testWindow"
+                    }
+                }
+            }
+        },
+
+        tester: {
+            type: "electron.tests.windowTester"
+        }
+    }
+});
+
+fluid.defaults("electron.tests.windowTester", {
+    gradeNames: "fluid.test.testCaseHolder",
+
+    modules: [
+        {
+            name: "Window creation tests",
+            tests: [
+                {
+                    expect: 1,
+                    name: "Bounds",
+                    sequence: [
+                        {
+                            event: "{app windough}.events.afterShow",
```

Comment:
Standard problem here - the afterShow event may fire synchronously before the sequence begins to bind to it. After we've merged the "new framework" impl we can improve the IoC Testing Framework so that it can win these construction races but in the meantime you will have to add some annoying boilerplate to every sequence to explicitly construct, e.g. the "windough" triggering an explicit "createOnEvent" for it.